tests/test_monitor.py

```diff
@@ -127,11 +127,25 @@ def test_new_clients_appear_in_focused_monitor(hlwm):
     assert hlwm.get_attr('clients', winid, 'tag') == 'tag2'
 
 
-def test_detect_monitors_does_not_crash(hlwm):
+@pytest.mark.parametrize("arg", ['-l', '--list-all', '--no-disjoin'])
+def test_detect_monitors_does_not_crash(hlwm, arg):
     # I don't know how to test detect_monitors properly, so just check that
     # it does not crash at least
+    hlwm.call(['detect_monitors', arg])
+
+
+def test_detect_monitors_root_window(hlwm):
+    list_monitors_before = hlwm.call('list_monitors').stdout
+    hlwm.call('add othertag')
+    hlwm.call('set_monitors 80x80+5+5 80x80+85+5')
+    assert hlwm.get_attr('monitors.count') == '2'
+
+    # check that detect monitors restores the one monitor
     hlwm.call('detect_monitors')
 
+    assert hlwm.get_attr('monitors.count') == '1'
```

Comment:
Could we make assumptions about the Xvfb monitor size and verify it here?

Reply:
This is what I initially intended to do (when writing the test case name). But then I decided to do the full testing in a separate PR (at least with Xephyr, one can simulate multiple screens).

Reply:
Alright, nice!

Reply:
Oh, and I totally missed that you basically already did what I had in mind by verifying that it matches the before-state. Sorry...